src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -74,8 +74,14 @@ export class TerminalService implements ITerminalService {
 	private _remoteTerminalsInitPromise: Promise<void> | undefined;
 	private _localTerminalsInitPromise: Promise<void> | undefined;
 	private _connectionState: TerminalConnectionState;
+	public get connectionState(): TerminalConnectionState { return this._connectionState; }
+	private readonly _localTerminalService?: ILocalTerminalService;
 
 	private _availableProfiles: ITerminalProfile[] | undefined;
+	public get availableProfiles(): ITerminalProfile[] {
+		this._refreshAvailableProfiles('external');
```

Comment:
Think we still want to throttle this, eg. if lots of things in the terminal access availableProfiles at once